File: node/enforced_sdk.js
const { init } = require("@osohq/express");
const express = require("express");

async function start() {
  const app = express();

  const apiKey =
    "<please provide your api key here>";
  const oso = init({
    apiKey,
    defaultActorId: (_req) => "anonymous",
  });

  app.get(
    "/organization/:org_id",
    oso.enforce({
      resourceId: ":org_id",
      action: "view",
      resourceType: "Organization",
    }),
    async (req, res) => {
      const orgId = req.params.org_id;
      res.status(200).send(`{"id": ${orgId}}`);
    }
  );
  app.listen(8000);
}

start();

File: python/enforced_sdk.py
import json

import oso_sdk
from flask import Flask
from oso_sdk.integrations.flask import FlaskIntegration

app = Flask(__name__)

api_key = "<please provide your api key here>"
with app.app_context():
    oso = oso_sdk.init(
        api_key,
        FlaskIntegration(),
    )


@oso.identify_user_from_request
def user() -> str:
    return "anonymous"


@app.route("/organization/<org_id>")
@oso.enforce("<org_id>", "view", "Organization")
def get(org_id):
    return json.dumps({"id": org_id}), 200


if __name__ == "__main__":
    app.run(port=8000)
